x-pack/plugins/security/public/account_management/account_management_app.tsx

```diff
@@ -96,7 +97,16 @@ export const Providers: FunctionComponent<ProvidersProps> = ({
         <I18nProvider>
           <KibanaThemeProvider theme$={theme$}>
             <Router history={history}>
-              <BreadcrumbsProvider onChange={onChange}>{children}</BreadcrumbsProvider>
+              <BreadcrumbsProvider onChange={onChange}>
+                <UserProfilesKibanaProvider
```

Comment:
The `<KibanaContextProvider>` already provides `CoreStart` contract to the render tree. 

Can you read that context instead of creating another provider component (`<UserProfilesKibanaProvider>`)?

Reply:
`KibanaContextProvider ` is a generic context provider. It can contain the `CoreStart` or maybe not. We can't rely on it. When exposing packages the recommended architecture is to provide your own context provider with the minimum required stateful Kibana dependencies.

Reply:
You can't rely on anything being available in React context since components can be rendered without the provider being present in the render tree. 

Creating unnecessary provider components clutters up the render tree and API without giving us any guarantees. 

I would recommend simplifying this and using the existing provider components if possible.

Reply:
This is not how the package architecture works. You have to expose the React context that you component/hook requires. You can't rely on anything being in the tree.

How it works is:
* Add the required context provider for your package
* Use the component / hook in the app (the hook internally can safely use the context it knows about)

If you need more info @clintandrewhall can help or point you to tech docs.

Reply:
You can safely import the required context provider and hook from  '@kbn/kibana-react-plugin/public' package inside `@kbn/user-profile-components`. 

There's no need to duplicate that code:

```
import { useKibana, KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
```

It works exactly the same way you're importing types, functions, components and hooks from EUI and core packages.

Reply:
Let's close this conversation as we simply having different developer preferences

I want to have:
- A clear, **unique** context with **specific** dependencies for the package
- A standalone package (think "npm") that does not depend on other context provider to be present in the tree 

I don't want:
- A big context provider that provides everything to the app and use TS generics to enforce type

You prefer the later, I prefer the former. No problem, let's move on 😊